File: py/features.py
# Nofrixion SDK feature factory

from feature.base_feature import NofrixionBaseFeature
from feature.test_feature import NofrixionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NofrixionBaseFeature(),
        "test": lambda: NofrixionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()

File: php/features.php
<?php
declare(strict_types=1);

// Nofrixion SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class NofrixionFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new NofrixionBaseFeature();
            case "test":
                return new NofrixionTestFeature();
            default:
                return new NofrixionBaseFeature();
        }
    }
}
